### ISSUE TITLE
alias file added to playlist automatically causes high CPU usage

### ISSUE THREAD

**Issue by mbzkr** [iina/iina#3411]
If there is an alias file in the same folder as currently playing video, when "Show Playlist Panel" is toggled, the CPU utilization spikes.  Even if video playback is stopped, CPU usage stays high until iina is restarted.  

After deleting the alias file, showing playlist panel does not cause high cpu.  

Issue is reproducible by creating an alias in the same dir as currently playing video and "Add files in the same folder automatically" to playlist is enabled in settings.

**Comment by low-batt** [comment]
With IINA 1.2.0 running under macOS 11.4 I setup a directory with a movie and an alias for that movie:
```text
low-batt@gag Movies$ ls -l
total 936
-rw-r--r--@ 1 low-batt  staff     844 Aug 14 23:34 1 alias.mp4
-rwxrwxrwx@ 1 low-batt  staff  474932 Jul 29 21:44 1.mp4
```

With IINA paused playing the movie and the playlist panel showing CPU usage was high as reported:
```text
low-batt@gag Movies$ while true; do ps -c -o 'comm,%cpu,cputime' -u low-batt | grep IINA | grep -v grep; sleep 15; done
IINA               0.0   0:00.01
IINA              21.1   0:00.38
IINA              68.9   0:02.75
IINA               3.2   0:04.09
IINA             125.1   0:05.60
IINA             123.3   0:23.42
IINA             134.0   0:39.72
IINA             113.4   0:56.07
^C
```

The mpv log contains this set of log messages repeated over and over:
```text
[  42.701][d][ffmpeg] AVIOContext: Statistics: 844 bytes read, 0 seeks
[  42.794][w][ffmpeg/demuxer] mov,mp4,m4a,3gp,3g2,mj2: Format mov,mp4,m4a,3gp,3g2,mj2 detected only with low score of 1, misdetection possible!
[  42.795][e][ffmpeg/demuxer] mov,mp4,m4a,3gp,3g2,mj2: moov atom not found
```

This is the same error that is reported by ffprobe if you have it probe the alias:
```text
low-batt@gag Movies$ ffprobe 1\ alias.mp4 
ffprobe version 4.4 Copyright (c) 2007-2021 the FFmpeg developers
  built with Apple clang version 12.0.5 (clang-1205.0.22.9)
  configuration: --prefix=/usr/local/Cellar/ffmpeg/4.4_2 --enable-shared --enable-pthreads --enable-version3 --cc=clang --host-cflags= --host-ldflags= --enable-ffplay --enable-gnutls --enable-gpl --enable-libaom --enable-libbluray --enable-libdav1d --enable-libmp3lame --enable-libopus --enable-librav1e --enable-librubberband --enable-libsnappy --enable-libsrt --enable-libtesseract --enable-libtheora --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxml2 --enable-libxvid --enable-lzma --enable-libfontconfig --enable-libfreetype --enable-frei0r --enable-libass --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-libspeex --enable-libsoxr --enable-libzmq --enable-libzimg --disable-libjack --disable-indev=jack --enable-avresample --enable-videotoolbox
  libavutil      56. 70.100 / 56. 70.100
  libavcodec     58.134.100 / 58.134.100
  libavformat    58. 76.100 / 58. 76.100
  libavdevice    58. 13.100 / 58. 13.100
  libavfilter     7.110.100 /  7.110.100
  libavresample   4.  0.  0 /  4.  0.  0
  libswscale      5.  9.100 /  5.  9.100
  libswresample   3.  9.100 /  3.  9.100
  libpostproc    55.  9.100 / 55.  9.100
[mov,mp4,m4a,3gp,3g2,mj2 @ 0x7ffad300c000] Format mov,mp4,m4a,3gp,3g2,mj2 detected only with low score of 1, misdetection possible!
[mov,mp4,m4a,3gp,3g2,mj2 @ 0x7ffad300c000] moov atom not found
1 alias.mp4: Invalid data found when processing input
```

A spindump shows DispatchQueue "IINAPlaylistTask" executing the method `PlaylistViewController.tableView`:
```text
8    closure #2 in PlaylistViewController.tableView(_:viewFor:row:) + 2828 (PlaylistViewController.swift:534,27 in IINA + 3799484) [0x10d1a99bc]
  8    PlayerCore.refreshCachedVideoInfo(forVideoPath:) + 211 (PlayerCore.swift:1637,39 in IINA + 4195491) [0x10d20a4a3]
    5    +[FFmpegController probeVideoInfoForFile:] + 70 (FFmpegController.m:331,33 in IINA + 33270) [0x10ce121f6]
```

This is another way to trigger the problem detailed in the response to issue #3162. The proposed fix in the pull request associated with that issue will correct this issue with high CPU usage as well.

That fix only addresses the excessive CPU usage. There may be other issues with aliases. As can be seen here mpv does not know how to open an alias to a file:
```text
low-batt@gag Movies$ /Applications/mpv.app/Contents/MacOS/mpv 1\ alias.mp4 
...
Failed to recognize file format.

Exiting... (Errors when loading file)
```

**Comment by low-batt** [comment]
The fix in PR #3477 has been merged into the develop branch.

**Comment by low-batt** [comment]
Closing. Fixed in IINA 1.3.0.
